server/athenian/api/cache.py

```diff
@@ -215,28 +215,38 @@ def cache_key(*args, **kwargs) -> bytes:
 def cached_methods(cls):
     """Decorate class to properly support cached instance methods."""
     for name, func in inspect.getmembers(cls, predicate=inspect.isfunction):
-        if getattr(func, "__cached__", False) and not getattr(func, "__cached_method__", False):
-            def generate_cls_method_shim(func):
-                def cls_method_shim(self):
-                    async def cached_wrapper(*args, **kwargs):
-                        return await func(self, *args, **kwargs)
+        outer_func = func
 
-                    async def reset_cache(*args, **kwargs):
-                        return await func.reset_cache(self, *args, **kwargs)
+        while not getattr(func, "__cached__", False) or getattr(func, "__cached_method__", False):
+            # This is needed in case the `@cached` decorator is not the outermost one
+            func = getattr(func, "__wrapped__", None)
+            if not func:
+                break
 
-                    @functools.wraps(func.cache_key)
-                    def cache_key(*args, **kwargs):
-                        return func.cache_key(self, *args, **kwargs)
+        if not func:
+            continue
```

Comment:
@vmarkovtsev PTAL at this. Basically, it's because of the usage of `@cached` as the non-outermost decorator as tested [here](https://github.com/athenianco/athenian-api/pull/1318/files#diff-13caa706ed0a55f3fb81caf5590fce9fc87f7c9e3f66a41ebd72b717872c8f7eR116).